src/engine/sidechain/sidechainworker.h

```diff
@@ -2,13 +2,67 @@
 #define SIDECHAINWORKER_H
 
 #include "util/types.h"
+#include "util/fifo.h"
+
+/*
+ * States:
+ * Error        Something errornous has happenned and can't go on
+ * New          First state before init
+ * Init         Initing state don't feed anything in this state
+ * Waiting      Waiting something not ready yet
+ * Busy         Is busy doing something can't process anything new
+ * Ready        Functioning ok
+ * Reading      Reading something and can't do anything else
+ * Writing      Writing something and can't do anything else
+ * Connected    Is connected to storage or server
+ * Connecting   Trying to connect storate or server
+ * Disconnected Ain't connected to storage or server
+ * 
+ * First state should be SIDECHAINWORKER_STATE_UNKNOWN and
+ * if state handling ain't supported by SideChainWorker-class
+ * then 'SIDECHAINWORKER_STATE_NEW' should be treated as
+ * SIDECHAINWORKER_STATE_READY. Newly written SideChainWorker-class
+ * should support state handling at leas this SIDECHAINWORKER_STATE_READY state.
+ */
+
+enum SidechaingStates {
```

Comment:
The main goal behind the shoutcast state CO is to support skins showing an 'on air' indicator light (maybe with a warning indicator, etc.).

The number of states here seems a little over-kill for achieving that goal.

Also no other sidechain worker than shoutcast has statuses or feedback like this, so I'm not convinced the generic name is more helpful here either. What other sidechains beyond broadcast do we imagine making use of these?